### PR TITLE
feat(premeeting/prejoin): Control roomname visibility

### PR DESCRIPTION
**Pull request by Calinteodor**
Proposal #14332

### CODE REVIEW

File: react/features/base/premeeting/components/web/PreMeetingScreen.tsx [hunk 1]
```diff
@@ -237,7 +240,8 @@ function mapStateToProps(state: IReduxState, ownProps: Partial<IProps>) {
             ? premeetingButtons
             : premeetingButtons.filter(b => isToolbarButtonEnabled(b, toolbarButtons)),
         _premeetingBackground: premeetingBackground,
-        _roomName: (hideConferenceSubject ? undefined : getConferenceName(state)) ?? ''
+        _roomName: (getFeatureFlag(state, MEETING_NAME_ENABLED, false)
```
Comment (saghul) [inline react/features/base/premeeting/components/web/PreMeetingScreen.tsx:243]:
```suggestion
        _roomName: (!getFeatureFlag(state, MEETING_NAME_ENABLED, true)
```

File: react/features/base/premeeting/components/web/PreMeetingScreen.tsx [hunk 2]
```diff
@@ -237,7 +240,8 @@ function mapStateToProps(state: IReduxState, ownProps: Partial<IProps>) {
...
+        && hideConferenceSubject ? undefined : getConferenceName(state)) ?? ''
```
Comment (saghul) [inline react/features/base/premeeting/components/web/PreMeetingScreen.tsx:244]:
```suggestion
        || hideConferenceSubject ? undefined : getConferenceName(state)) ?? ''
```

File: react/features/prejoin/components/native/Prejoin.tsx [hunk 3]
```diff
@@ -58,6 +61,8 @@ const Prejoin: React.FC<IPrejoinProps> = ({ navigation }: IPrejoinProps) => {
         = useSelector((state: IReduxState) => !getFeatureFlag(state, PREJOIN_PAGE_HIDE_DISPLAY_NAME, false));
     const isDisplayNameReadonly = useSelector(isNameReadOnly);
     const roomName = useSelector((state: IReduxState) => getConferenceName(state));
+    const roomNameEnabled = useSelector((state: IReduxState) => getFeatureFlag(state, MEETING_NAME_ENABLED, true)
```
Comment (saghul) [inline react/features/prejoin/components/native/Prejoin.tsx:64]: Factor out the selector and use it in both places.

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/prejoin/functions.ts [hunk 4]
```diff
@@ -184,3 +187,14 @@ export function isUnsafeRoomWarningEnabled(state: IReduxState): boolean {
...
     return getFeatureFlag(state, UNSAFE_ROOM_WARNING, enableInsecureRoomNameWarning);
 }
+
+/**
+ * Returns true if the room name is enabled.
+ *
+ * @param {IReduxState} state - The state of the app.
+ * @returns {boolean}
+ */
+export function isRoomNameEnabled(state: IReduxState): boolean {
```
Comment (saghul) [inline react/features/prejoin/functions.ts:197]: Pl use this in web too.

Comment (Calinteodor) [reply react/features/prejoin/functions.ts:197]: Done

**Review by saghul — APPROVED** [review]
LGTM. The iOS failure is due to the CI, I'll fix that.

**Pull request merged**